File: raceClient/KataBattles/RomanNumeralsKataRuby/spec/romannumerals_spec.rb
#romannumerals_spec
require './lib/romannumerals'

describe "Roman Numerals" do

  before(:each) do
    @roman = RomanNumeral.new
  end


  context "Converting Roman Numerals to number - " do
    it "should convert I to 1" do
      @roman.convert("I").should == 1
    end

    it "should convert II to 2" do
      @roman.convert("II").should == 2
    end

    it "should convert III to 3" do
      @roman.convert("III").should == 3
    end

    it "should convert IV to 4" do
      @roman.convert("IV").should == 4
    end

    it "should convert V to 5" do
      @roman.convert("V").should == 5
    end

    it "should convert VI to 6" do
      @roman.convert("VI").should == 6
    end

    it "should convert VII to 7" do
      @roman.convert("VII").should == 7
    end

    it "should convert VIII to 8" do
      @roman.convert("VIII").should == 8
    end

    it "should convert IX to 9" do
      @roman.convert("IX").should == 9
    end

    it "should convert X to X" do
      @roman.convert("X").should == 10
    end


  end
end

File: raceClient/KataBattles/RomanNumeralsJS/spec/accept/RomanNumeralsSpec.js
// Acceptance tests
describe('Roman Numerals', function() {
  //var romanNumerals = require('./lib/rn.js');
  var numerals;

  beforeEach(function() {
    numerals = new romanNumerals();
  });

  it('translates I to 1', function(){
  	expect(numerals.convert('I')).toEqual(1)
  });

  it('translates II to 2', function(){
  	expect(numerals.convert('II')).toEqual(2)
  });

  it('translates III to 3', function(){
  	expect(numerals.convert('III')).toEqual(3)
  });

  it('translates IV to 4', function(){
  	expect(numerals.convert('IV')).toEqual(4)
  });

  it('translates V to 5', function(){
  	expect(numerals.convert('V')).toEqual(5)
  });

  it('translates VI to 6', function(){
  	expect(numerals.convert('VI')).toEqual(6)
  });

  it('translates VII to 7', function(){
  	expect(numerals.convert('VII')).toEqual(7)
  });

  it('translates VIII to 8', function(){
  	expect(numerals.convert('VIII')).toEqual(8)
  });

  it('translates IX to 9', function(){
  	expect(numerals.convert('IX')).toEqual(9)
  });

  it('translates X to 10', function(){
  	expect(numerals.convert('X')).toEqual(10)
  });

});
